### DISCUSSION TITLE
Issues installing directus with pnpm

### DISCUSSION THREAD

**Discussion by flayks** [directus/directus#11619] [Feature Requests]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

Although the installation seems to be working with pnpm, it returns some peer dependencies issues. This seems to be happening since 9.5.0 or 9.4.0 as it was working fine since 9.0

### To Reproduce

Use `pnpm install`

### Errors Shown

```
dependencies:
+ directus 9.5.1
+ mysql 2.18.1

 WARN  Issues with peer dependencies found
.
└─┬ directus
  └─┬ @directus/extensions-sdk
    ├─┬ @directus/shared
    │ ├── ✕ missing peer @types/express@"*"
    │ ├── ✕ missing peer @types/geojson@"*"
    │ ├── ✕ missing peer @types/pino@"*"
    │ ├── ✕ missing peer geojson@"*"
    │ ├── ✕ missing peer vue@3
    │ ├── ✕ missing peer vue-i18n@9
    │ └── ✕ missing peer vue-router@4
    └─┬ rollup-plugin-typescript2
      ├── ✕ missing peer typescript@>=2.4.0
      └─┬ @yarn-tool/resolve-package
        ├── ✕ missing peer @types/node@"*"
        ├─┬ @ts-type/package-dts
        │ ├── ✕ missing peer @types/bluebird@"*"
        │ ├── ✕ missing peer @types/node@"*"
        │ ├── ✕ missing peer ts-toolbelt@"*"
        │ └─┬ ts-type
        │   ├── ✕ missing peer @types/bluebird@"*"
        │   ├── ✕ missing peer @types/node@"*"
        │   └── ✕ missing peer ts-toolbelt@^9.6.0
        └─┬ upath2
          └── ✕ missing peer @types/node@"*"
Peer dependencies that should be installed:
  @types/bluebird@"*"            @types/geojson@"*"             @types/pino@"*"                ts-toolbelt@">=9.6.0 <10.0.0"  vue-i18n@9                     vue@3                          
  @types/express@"*"             @types/node@"*"                geojson@"*"                    typescript@>=2.4.0             vue-router@4                   
```

### What version of Directus are you using?

9.5.1

### What version of Node.js are you using?

16

### What database are you using?

MySQL

### What browser are you using?

Brave

### What operating system are you using?

macOS

### How are you deploying Directus?

locally

**Comment by oreilles** [comment]
While supporting pnpm would be nice, and is actually in discussion considering issues with npm, it is not the case at the moment.
Since this issue is not related to a bug with Directus but rather a feature request, I'm going to move it to a discussion. 👍

**Comment by flayks** [comment]
@Oreilles I was just wondering why it was working perfectly until version 9.5? Did something major changed in between?

**Comment by azrikahar** [comment]
@flayks I believe the "change" in question is related to #11231 (and possibly other related PRs in `9.5.0` & `9.5.1`) where type deps were moved to `peerDependencies`. 

Granted this is just from a google search & I don't have personal experience with pnpm, I believe pnpm does not auto installs peer dependencies like npm. This thread, specifically this fairly recent answer: https://github.com/pnpm/pnpm/discussions/3995#discussioncomment-1958472 does point out that you should be able to enable the `auto-install-peers` flag for it to behave like npm which should solve your "Peer dependencies that should be installed" issue.

**Comment by rijkvanzanten** [comment]
(For the record, this is regarding installing the repo from source, not installing Directus as a regular package into another project)

**Reply by Nisthar** [reply]
> (For the record, this is regarding installing the repo from source, not installing Directus as a regular package into another project)

Does it work if you are just installing (not from the source)?

**Reply by rijkvanzanten** [reply]
Yes, should be fine 👍🏻

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
